src/js/node/fs.promises.ts

```diff
@@ -586,22 +586,17 @@ async function writeFileAsyncIteratorInner(fd, iterable, encoding) {
   const mustRencode = !(encoding === "utf8" || encoding === "utf-8" || encoding === "binary" || encoding === "buffer");
   let totalBytesWritten = 0;
 
-  try {
-    for await (let chunk of iterable) {
-      if (mustRencode && typeof chunk === "string") {
-        $debug("Re-encoding chunk to", encoding);
-        chunk = Buffer.from(chunk, encoding);
-      }
+  for await (let chunk of iterable) {
+    if (mustRencode && typeof chunk === "string") {
+      $debug("Re-encoding chunk to", encoding);
+      chunk = Buffer.from(chunk, encoding);
+    }
+    totalBytesWritten += chunk.length;
 
-      const prom = writer.write(chunk);
-      if (prom && $isPromise(prom)) {
-        totalBytesWritten += await prom;
-      } else {
-        totalBytesWritten += prom;
-      }
+    const prom = writer.write(chunk);
+    if (prom && $isPromise(prom)) {
+      await prom;
     }
-  } finally {
-    await writer.end();
```

Comment:
`end` should always be called when the writer is done

this would be masking a bug that only occurs in `end`